Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XldbDataStore.cs

```diff
@@ -68,14 +68,133 @@ public IEnumerable<string> GetEventsByType(ExecutionEventId eventTypeID)
+                                break;
+                            case ExecutionEventId.DirectoryMembershipHashed:
+                                storedEvents.Add(DirectoryMembershipHashedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ProcessExecutionMonitoringReported:
+                                storedEvents.Add(ProcessExecutionMonitoringReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ProcessFingerprintComputation:
+                                storedEvents.Add(ProcessFingerprintComputationEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ExtraEventDataReported:
+                                storedEvents.Add(ExtraEventDataReported.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.DependencyViolationReported:
+                                storedEvents.Add(DependencyViolationReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipExecutionStepPerformanceReported:
+                                storedEvents.Add(PipExecutionStepPerformanceReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipCacheMiss:
+                                storedEvents.Add(PipCacheMissEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ResourceUsageReported:
+                                storedEvents.Add(StatusReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.DominoInvocation:
+                                storedEvents.Add(BXLInvocationEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipExecutionDirectoryOutputs:
+                                storedEvents.Add(PipExecutionDirectoryOutputsEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            default:
+                                break;
+                        }
                     }
                 })
             );
 
             return storedEvents;
         }
 
+        /// <summary>
+        /// Gets all the File Artifact Content Decided Events
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<string> GetFileArtifactContentDecidedEvents() => GetEventsByType(ExecutionEventId.FileArtifactContentDecided);

```

Comment:
Do you really need all these helper properties?  I'd guess that just exposing `GetEventsByType` is enough.

Reply:
Initial plan was to just expose GetEventsByType and not have any of these other methods, but then we later thought that these specific endpoints would allow users more easily get the information they were looking for (instead of trying to fiddle around with the right enum values to pass into GetEventsByType)